src/results.rs

```diff
@@ -71,3 +72,94 @@ pub(crate) struct GetMoreResult {
     pub(crate) batch: VecDeque<Document>,
     pub(crate) exhausted: bool,
 }
+
+#[derive(Debug, Clone, Serialize, PartialEq)]
+#[serde(rename_all = "lowercase", untagged)]
+#[non_exhaustive]
+/// Describes the type of data store returned when executing
+/// [`Database::list_collections`](../struct.Database.html#method.list_collections).
+pub enum CollectionType {
+    /// Indicates that the data store is a view.
+    View,
+
+    /// Indicates that the data store is a collection.
+    Collection,
+
+    /// An unknown collection type. This is included for forwards compatibility.
+    Other(String),
+}
+
+impl<'de> Deserialize<'de> for CollectionType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de> {
+        let s = String::deserialize(deserializer)?;
+        let out = match s.as_str() {
+            "collection" => CollectionType::Collection,
+            "view" => CollectionType::View,
+            _ => CollectionType::Other(s)
+        };
+        Ok(out)
+    }
+}
+
+/// Info about the collection that is contained in the `CollectionSpecification::info` field of a specification returned from
+/// [`Database::list_collections`](../struct.Database.html#method.list_collections).
+///
+/// See the MongoDB [manual](https://docs.mongodb.com/manual/reference/command/listCollections/#listCollections.cursor)
+/// for more information.
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+pub struct CollectionSpecificationInfo {
+    /// Indicates whether or not the data store is read-only.
+    pub read_only: bool,
+
+    /// The collection's UUID - once established, this does not change and remains the same across replica
+    /// set members and shards in a sharded cluster. If the data store is a view, this field is `None`.
+    pub uuid: Option<Binary>,
+}
+
+#[derive(Debug, Clone, Deserialize, Serialize)]
+#[serde(rename_all = "camelCase")]
+#[non_exhaustive]
+/// Information about a collection as reported by [`Database::list_collections`](../struct.Database.html#method.list_collections).
+pub struct CollectionSpecification {
+    /// The name of the collection.
+    pub name: String,
+
+    /// Type of the data store.
+    #[serde(rename="type")]
+    pub collection_type: CollectionType,
+
+    /// The options used to create the collection.
+    pub options: CreateCollectionOptions,
```

Comment:
I thought about creating a separate type for this in case they ever diverge (and also because it contains fields like `write_concern` that don't make sense in this context), but the [docs](https://docs.mongodb.com/manual/reference/command/listCollections/#output) explicitly state that this field will contain the exact options provided to `createCollection`, and looking through the server code confirms also this, so I figured this was pretty low risk. Let me know what you think.

Reply:
I agree it seems low risk